projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.ts

```diff
@@ -169,15 +178,53 @@ export class ProductListComponentService {
       .unsubscribe();
   }
 
+  /**
+   * Sort the search results by the given sort code.
+   */
   sort(sortCode: string): void {
-    this.setQueryParams({ sortCode });
+    this.route({ sortCode });
   }
 
-  private setQueryParams(queryParams: SearchCriteria): void {
+  /**
+   * Routes to the next product listing page, using the given `queryParams`. The
+   * `queryParams` support sorting, pagination and querying.
+   *
+   * The `queryParams` are delegated to the Angular router `NavigationExtras`.
+   */
+  protected route(queryParams: SearchCriteria): void {
     this.router.navigate([], {
       queryParams,
       queryParamsHandling: 'merge',
       relativeTo: this.activatedRoute,
     });
   }
+
+  /**
+   * The site context is exposed to update the search query in case of a
+   * changing context. The context will typically influence the search data.
+   *
+   * We keep this private for now, as we're likely refactoring this in the next
+   * major version.
+   */
+  private get siteContext(): Observable<any>[] {
```

Comment:
`Observable<string>`? 🙏🏻

Reply:
FYI as a result i had to refactor the code, as the `pluck` couldn't figure out the different types (RouterState vs string). Pluck seems to be dropped anyway in rxjs.

Reply:
Good to know 👍🏻